packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmImpl.kt

```diff
@@ -50,7 +50,9 @@ import kotlin.reflect.KClass
 // TODO API-PUBLIC Document platform specific internals (RealmInitializer, etc.)
 internal class RealmImpl private constructor(
     configuration: InternalConfiguration,
-    dbPointer: NativePointer
+    // TODO Should actually be a frozen pointer, but since we cannot directly obtain one we expect
+    //  a live reference and grab the frozen version of that in the init-block
+    dbPointer: LiveRealmPointer,
```

Comment:
Great that this is now captured by the type system 👍

Reply:
Yep, should help avoiding calling all the update methods on a frozen realm pointer, etc. ... if only the IDE was actually able to recognize type symbols and types 😞